_posts/2019-03-17-importance-of-using-linters-and-metrics-in-ruby-copy.markdown

```diff
@@ -0,0 +1,122 @@
+
+I've already discussed the ad revenue model. Well, Google thrives on this model. It reportedly earns [86% of it's revenue from ads last quarter](https://www.theverge.com/2018/10/25/18024654/alphabet-google-q3-2018-earnings-ad-business-booming). There is growing discontentment with this model. Creators revenue is falling and consumers are increasingly using adblockers to avoid annoying ads and protect their privacy. The use of adblockers is against Google's interest so they will be forced to act against it.
+
+Google recently announced that they would [start blocking ads](https://www.wired.com/story/google-chrome-ad-blocker-change-web/). However, the details reveal that they're only blocking sites which have clearly disruptive and obnoxious ads, Wired estimated about 100,000 site in total. What Google are really trying to do is block just enough ads to stop the uptake of third party adblockers.
+
+### Speed and Performance
+Brave is reportedly faster and consumes less memory than Chrome so I wanted to put that to the test.
+
+**Load speed**
+
+I tested the loading speed of Brave and Chrome across 3 different sites; YouTube, Amazon & The Guardian. Load times were 21-27% faster on Brave than Chrome. I assume this is because Brave's adblocker is stricter and is blocking more network calls.
+
+On The Guardian's site, for example, you can see that Brave reports that it blocked 17 items where ABP only blocked 8.
+
+![abp-guardian-website](/assets/abp-guardian-website.png){:.image-50}
+
+![brave-guardian-website](/assets/brave-guardian-website.png){:.image-50}
+
+When I looked at the network traffic Brave made 178 requests and Chrome, with ABP, made 238 requests.
+
+![abp-guardian-website-network-traffic](/assets/abp-guardian-website-network-traffic.png){:.image-50}
+
+![brave-guardian-website-network-traffic](/assets/brave-guardian-website-network-traffic.png){:.image-50}
+
+TODO: zoom in on the important details here
+
+With a fast internet connection, 20-30% speed improvement is not particularly noticeable. But for users with slower connections the difference could be noticeable.
+
+**Memory consumption**
+
+I tested the memory consumption of Brave, Chrome and Chromium. I tested them with 0 tabs, 1 tab, 2 tabs, 6 tabs, 10 tabs and 20 tabs. The tests were conducted 3 times each.
+
+![brave-chrome-chromium-memory-consumption](/assets/brave-chrome-chromium-memory-consumption.png)
+
+Chrome and Brave have the same resting (0 tabs) memory usage. As the number of tabs increase, Brave outperforms. Above 6 tabs Brave uses 20-30% less memory. I haven't found any evidence to explain why Brave has better memory usage, given that both browsers are based off Chrome. I can only guess that Chrome has more bloatware.
```

Comment:
> I can only guess that Chrome has more bloatware.

It's extensions and having more stuff blocked by default which reduces the memory usage.

Reply:
Do you mean Chrome has more extension installed by default?

Reply:
@tomkadwill No, people often just install lots of rubbish ones, which try to do a lot of each page load.